### PR TITLE
[WIP] php5 compatibility

### PR DESCRIPTION
**Pull request by maglnet**

### CODE REVIEW

**Comment by malukenho** [comment]
Why this downgrade?

File: example/test.php [hunk 1]
```diff
@@ -10,24 +10,22 @@
 
 // Please run "run-test.sh", and not this file directly.
 
-(function () {
```
Comment (Ocramius) [inline example/test.php]: Please restore the previous approach, but use `call_user_func(function () { ... });`

File: src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php [hunk 2]
```diff
@@ -6,7 +6,12 @@
 
 final class LocateAllFilesByExtension
 {
-    public function __invoke(Traversable $directories, string $fileExtension) : Traversable
+    /**
+     * @param Traversable $directories
+     * @param string $fileExtension
+     * @return \Traversable
+     */
+    public function __invoke(Traversable $directories, string $fileExtension)
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php]: `string` is not a valid hint in 5.x

File: src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php [hunk 3]
```diff
@@ -35,22 +56,38 @@ function (string $sourceDir) use ($packageDir) {
         ));
     }
 
-    private function locateFilesInPsr0Definitions(array $locations) : Generator
+    /**
+     * @param array $locations
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: `string[]` (also the others)

**Comment by Ocramius** [comment]
@malukenho I built this component mainly as a PoC, but @maglnet is taking over (I just got it kickstarted)

**Comment by Ocramius** [comment]
@maglnet actually, indeed, here's a question: since this tool will *not* be part of production environments, but only dev environments, is there a reason not to simply use PHP7?

Consider that nobody should really require this package in any app...

Basically, any environment can run `/path/to/php7 composer-require-checker/check.php path/to/composer.json`

**Comment by maglnet** [comment]
@Ocramius, @malukenho 
the main problem is our build environment, which doesn't provide php7 for now.

regarding the massive use of generators and the complexity to rewrite these, I'll think about a different approach to run it within our companies environment.

**Comment by jk** [comment]
@maglnet the release date of php7 isn't too far away, is it? I'll probably make the final available than.

**Comment by maglnet** [comment]
Then I should stop working on this. ;)

**Comment by Ocramius** [comment]
That's what I was suggesting - for a build environment, you could even just compile 7 (takes 10 minutes) and then use that just for this task :-)

**Comment by maglnet** [comment]
So, I'll close this one, we'll try to get php7 running on our build server, as soon as it's released.

**Pull request closed without merging**